source/particles/particle_handler.cc

```diff
@@ -837,7 +837,7 @@ namespace Particles
                                        recv_data_it);
       }
 
-    AssertThrow(recv_data_it == &recv_data.back()+1,
+    AssertThrow(total_recv_data == 0 || recv_data_it == &recv_data.back()+1,
```

Comment:
I believe the second clause is technically correct, but is there any reason why we did not write `recv_data.end()`?

Reply:
`recv_data_it` is a (`const void *`) pointer while `recv_data.end()` returns an iterator. I am not sure if there is a better way to compare them.

Reply:
How about using `recv_data.data() + recv_data.size()`? That's what the assertion wants to test: that we've traversed through the entire `recv_data` array and have handled all of its content.

Reply:
This is consistent with what we are doing elsewhere.